src/System.Net.Security/src/System/Net/Unix/SslConnectionInfo.cs

```diff
@@ -92,8 +105,13 @@ private static HashAlgorithmType MapHashAlgorithmType(long mac)
                     return HashAlgorithmType.Md5;
                 case 2:
                     return HashAlgorithmType.Sha1;
+                case 4:
+                case 8:
+                case 16:
+                case 32:
+                case 64:
+                    return (HashAlgorithmType) mac;
```

Comment:
Same as MapCipherAlgorithmType. We need to not have two different unmapped values for the same thing.